File: 751-1000/868. Binary Gap/solution.cpp
class Solution {
public:
    int binaryGap(int n) {
        int ans = 0 ; 
        int prev = -1 ; 
        int curr = 0 ; 
        while( n > 0 ){
            if( n % 2 == 1 ){
                if( prev != -1 ){
                    ans = max( ans , curr - prev ) ; 
                }
                prev = curr ; 
            }
            n >>= 1 ; 
            curr++ ; 
        }
        return ans ;  
    }
};

File: 751-1000/868. Binary Gap/solution.py
class Solution:
    def binaryGap(self, n: int) -> int:
        ans = 0 
        prev = -1 
        curr = 0 
        while n : 
            if n % 2 == 1 : 
                if prev != -1 : 
                    ans = max( ans , curr - prev )
                prev = curr 
            n >>= 1 
            curr += 1 
        return ans 